src/synthesized_insight/metrics/base.py

```diff
@@ -83,6 +83,20 @@ def __call__(self, sr_a: pd.Series, sr_b: pd.Series) -> Tuple[Union[int, float,
         return self._compute_test(sr_a, sr_b)
 
 
+class DataFrameMatrix(_Metric):
+    """Operates on columns of a single DataFrame. Returns results in DataFrame/s matrix"""
+    @abstractmethod
+    def __call__(self, df: pd.DataFrame) -> Union[pd.DataFrame, Tuple[pd.DataFrame, Union[pd.DataFrame, None]]]:
```

Comment:
The return type here looks pretty complicated. I suppose its because sometimes we are returning a dataframe of values and sometimes a dataframe of **p_values as well** right?

It would be nice if we could keep the return types as just a single DataFrame.

Maybe using something like this could help?

```pycon

>>> import pandas as pd
>>> import numpy as np
>>> values = np.array([1, 1.5, 2])
>>> p_values = np.array([0.5, 0.1, 0.01])

>>> print(values, values.dtype)
[1.  1.5 2. ] float64

>>> print(p_values, p_values.dtype)
[0.5  0.1  0.01] float64

>>> metric = np.dtype([
...     ('value', 'f4'),
...     ('p_value', 'f4')
... ])
>>> combined = np.array([a  for a in zip(values, p_values)], dtype=metric)

>>> print(combined, combined.dtype)
[(1. , 0.5 ) (1.5, 0.1 ) (2. , 0.01)] [('value', '<f4'), ('p_value', '<f4')]

>>> print(combined['value'], 
[1.  1.5 2. ]

>>> print(combined['p_value'])
[0.5  0.1  0.01]
```

Reply:
If we want to keep it simple then we could just set the type as 
`Tuple[pd.DataFrame, Optional[pd.DataFrame]]`
which is a bit nicer than what you have here

Reply:
I think `Tuple[pd.DataFrame, Optional[pd.DataFrame]]` is better for the return type, hence changed it so.

Reply:
Yeah makes sense! Lets have a think about how we can improve this moving forward though